ext/intl/locale/locale_methods.c

```diff
@@ -1080,12 +1080,13 @@ static int add_array_entry(const char* loc_name, zval* hash_arr, char* key_name)
 			if( cur_key_name ){
 				efree( cur_key_name);
 			}
-			cur_key_name = (char*)ecalloc( 25,  25);
-			sprintf( cur_key_name , "%s%d", key_name , cnt++);
+			size_t cur_key_name_size = 11 + strlen(key_name) + 1;
```

Comment:
I'm confused by how you computed this length

Reply:
-2147483648 has length 11, plus string length, plus NUL
By accounting for the longest possible integer we avoid reallocating

Reply:
Ah right, maybe just add a comment about this?